plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java

```diff
@@ -166,11 +167,11 @@ public void test_invalid_settings_to_create_azure_repository() {
 
     private String httpServerUrl() {
         InetSocketAddress address = httpServer.getAddress();
-        return "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
+        return "http://" + address.getAddress().getCanonicalHostName() + ":" + address.getPort();
     }
 
     private String invalidHttpServerUrl() {
         InetSocketAddress address = httpServer.getAddress();
-        return "http://dummy.invalid:" + address.getPort();
+        return "http://invalid." + address.getAddress().getCanonicalHostName() + ":" + address.getPort();
```

Comment:
Is this change really related? As this will be invalid anyhow, I don't quite understand what this change improves...

Reply:
The secondary endpoint must be a subdomain of the primary.
So when the primary is an IP then this ^^ validation passes somehow, but then it seems for ci it frequently fails afterwards to connect to the http server.
With the change to `localhost` the secondary endpoint must be` <something>.localhost` to not trigger the validation, I will clarify it in the commit msg/PR description.

thank you!